### PR TITLE
fix: template literal expression having a non-string type.

### LINKED ISSUE #2322: (JS-0378) Found template literal expression having a non-string type
## Description
Using only string values in placeholders is recommended as otherwise, the value may be wrongly displayed; for example, if an object value is directly included in a string, it will be shown as `[object Object]`. Using only string values can also ensure that `null` or `undefined` values are not directly …

## Occurrences
There are 23 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/kodadot/nft-gallery/issue/JS-0378/occurrences/](https://deepsource.io/gh/kodadot/nft-gallery/issue/JS-0378/occurrences/)

image

### PR DESCRIPTION
**Pull request by zhyd1997**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #2322
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=H2DifyJqaansQREdGFU4BxZyaApbkCgP1a9Pq4xKaebSFZh)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: utils/filters.ts [hunk 1]
```diff
@@ -14,14 +14,13 @@ export const toPercent = (value: number | undefined | BN): string => {
   }
 
   if (isNumber(value)) {
-    return Number.isInteger(value) ? `${value}.00 %` : `${value} %`
+    const floatNum = Number.parseFloat(value.toString()).toFixed(2)
+    return `${floatNum} %`
   }
 
-  if (BN.isBN(value)) {
```
Comment (zhyd1997) [inline utils/filters.ts:20]: **Note**: some logic changed here.

`value` type has explicitly declared as `number`, `undefined` and `BN`, we don't need to check again.

here, it's type is `BN` only.

**Comment by zhyd1997** [comment]
@roiLeo If you have any questions, please let me know.

**Comment by yangwao** [comment]
Yup, @roiLeo let's check this one, if it's all good 😊

File: utils/filters.ts (hunk 1 above)
Comment (yangwao) [reply utils/filters.ts:20]: Maybe @vikiival can take a peak 😇

Comment (roiLeo) [reply utils/filters.ts:20]: this filter is not used for now

File: utils/ipfs.ts [hunk 2]
```diff
@@ -41,8 +41,10 @@ export const ipfsToArweave = async (ipfsLink: string): Promise<string> => {
     }
 
     return ''
-  } catch (e: any) {
-    console.error(`[IPFS2AR] Unable to Arweave ${e.message}`)
+  } catch (e: unknown) {
```
Comment (roiLeo) [inline utils/ipfs.ts:44]: why did you change `any` here?

**Review by roiLeo — COMMENTED** [review]
Lgtm, some of the `skipcq` are not even used anymore
Is it really necessary to have added date-fns import into indexer file? Does it fix something?

File: utils/filters.ts (hunk 1 above)
Comment (vikiival) [reply utils/filters.ts:20]: > type has explicitly declared

It could still fail in runtime.

> this filter is not used for now

I would keep it for future royalties component

**Review by vikiival — CHANGES_REQUESTED** [review]
check

`import { logError } from '@/utils/mappers'`

**Comment by zhyd1997** [comment]
@yangwao @roiLeo @vikiival 

Thank you a lot 🙏
I will make theses changes:
1. Remove `BN.isBN()` filter.
2. and try `logError`.

File: utils/filters.ts (hunk 1 above)
Comment (zhyd1997) [reply utils/filters.ts:20]: @riton @vikiival 
kept it!

File: utils/ipfs.ts [hunk 3]
```diff
@@ -42,7 +44,9 @@ export const ipfsToArweave = async (ipfsLink: string): Promise<string> => {
 
     return ''
   } catch (e: any) {
```
Comment (zhyd1997) [inline utils/ipfs.ts:46]: @roiLeo revert to `any` type
@vikiival use `logError` here

File: store/indexer.ts [hunk 4]
```diff
@@ -38,10 +39,15 @@ export const actions: ActionTree<IndexerState, IndexerState> = {
...
+      const lastProcessedTime = formatDistanceToNow(
+        new Date(Number(data?.lastProcessedTimestamp)),
+        { addSuffix: true }
+      )
+
       console.log(
         `%cIndexer:
         Health: ${data?.indexerHealthy ? '❤️' : '💀'}
-        Last: ${new Date(Number(data?.lastProcessedTimestamp))}`,
+        Last: ${lastProcessedTime}`,
```
Comment (zhyd1997) [inline store/indexer.ts:50]: > Is it really necessary to have added date-fns import into indexer file? Does it fix something?

@roiLeo 
original string literal is `Date` type, so I replaced it with string message.

File: utils/filters.ts (hunk 1 above)
Comment (zhyd1997) [reply utils/filters.ts:20]: @vikiival @roiLeo 
kept the filter finally.

**Comment by yangwao** [comment]
Seems @zhyd1997 you've got a different handle in another git client? ;) 
image

**Comment by zhyd1997** [comment]
> Seems @zhyd1997 you've got a different handle in another git client? ;) image

Yes, because my gitpod free 50 hours ran out, so I log in with another github account.

`git push` is ok but not `git rebase` and then `git push --force` (displayed new github username here).

It's my fault, maybe I need to update the branch directly on this PR with the CTA button like so: https://twitter.com/github/status/1489672242074206209?s=20&t=7tT_tnB41xaPup730K7rEA.

**Comment by yangwao** [comment]
> > Seems @zhyd1997 you've got a different handle in another git client? ;) image
> 
> Yes, because my gitpod free 50 hours ran out, so I log in with another github account.
> 
> `git push` is ok but not `git rebase` and then `git push --force` (displayed new github username here).
> 
> It's my fault, maybe I need to update the branch directly on this PR with the CTA button like so: [twitter.com/github/status/1489672242074206209?s=20&t=7tT_tnB41xaPup730K7rEA](https://twitter.com/github/status/1489672242074206209?s=20&t=7tT_tnB41xaPup730K7rEA).

Ah I see, updated it for ya! :)

**Comment by vikiival** [comment]
Why rebase?

**Comment by zhyd1997** [comment]
@vikiival because the branch is behind the origin `main` branch.

**Comment by zhyd1997** [comment]
@vikiival what I do is the same as below, but I forgot, it's my mistake:
update branch

File: components/rmrk/utils.ts [hunk 5]
```diff
@@ -391,7 +395,9 @@ export const isJsonGltf = (value: any): boolean => {
 
     return true
   } catch (e) {
-    console.warn(`Unable to decide on isJsonGltf ${e}`)
+    if (e instanceof Error) {
+      console.warn(`Unable to decide on isJsonGltf ${e.message}`)
+    }
```
Comment (vikiival) [inline components/rmrk/utils.ts:400]: and others ?

**Review by vikiival — CHANGES_REQUESTED** [review]
There are still some code snippets which can be deduplicated

**Comment by vikiival** [comment]
>  because the branch is behind the origin main branch.

#2372

**Comment by zhyd1997** [comment]
@vikiival Done: ba14fb0

also sorry for `rebase` branch, I prefer using it to prevent polluting the git history, I will do `merge` next time.

**Comment by zhyd1997** [comment]
Closed it because I used `git rebase` before, I will submit a new PR.
